pkg/scheduler/cache/cache.go

```diff
@@ -317,6 +317,7 @@ func (cache *schedulerCache) UpdatePod(oldPod, newPod *v1.Pod) error {
 		if err := cache.updatePod(oldPod, newPod); err != nil {
 			return err
 		}
+		currState.pod = newPod
```

Comment:
Would you share the case that this PR handled? And as we're changed to use pod's UID as key, that may make cache in-consistent.

Reply:
oh, sorry for confusion: the UID did not change when updating pod. so the fix seems ok. Let me check it today :)

Reply:
@k82cn I think this PR just makes sure the cache always keep consistent. We won't have any problem now if we don't update the podStates map, but if pod resources could be updated(like in-place pod resource update), this will bring problem.

Reply:
Yes, resource update is one case. Another case is `GetPod`, which may return 'old' pod; can you help to add a test for it? For assumed pod, `GetPod` will return the correct value; but for assigned pods, it may return old pod (although no one uses this feature right now).

Reply:
sure I will add a test case asap.